### PR TITLE
WIP: Detect GCC/Mingw64 in PATH when running in Windows command prompt

### PR DESCRIPTION
**Pull request by rongjiecomputer**
When running in MSYS

```
bazel build --compiler=mingw-gcc # Use <MSYS_ROOT>/mingw64/bin/gcc
bazel build --compiler=msys-gcc # Use <MSYS_ROOT>/usr/bin/gcc
```

When running in Windows command prompt

```
# Use gcc from PATH if found, else use <MSYS_ROOT>/mingw64/bin/gcc
bazel build --compiler=mingw-gcc 
bazel build --compiler=msys-gcc # Use <MSYS_ROOT>/usr/bin/gcc
```

MSVC will still be selected when running without `--compiler` or with `--compiler=msvc-cl`.

/cc @meteorcloudy

### CODE REVIEW

File: tools/cpp/windows_cc_configure.bzl [hunk 1]
```diff
@@ -308,8 +325,8 @@ def configure_windows_toolchain(repository_ctx):
                 "%{dbg_mode_debug}": "/DEBUG",
                 "%{fastbuild_mode_debug}": "/DEBUG",
                 "%{compilation_mode_content}": "",
-                "%{content}": _get_escaped_windows_msys_crosstool_content(repository_ctx),
-                "%{msys_x64_mingw_content}": _get_escaped_windows_msys_crosstool_content(repository_ctx, use_mingw = True),
+                "%{content}": _get_escaped_windows_msys_crosstool_content(repository_ctx, "mingw"),
+                "%{msys_x64_mingw_content}": _get_escaped_windows_msys_crosstool_content(repository_ctx, "msys"),
                 "%{opt_content}": "",
```
Comment (meteorcloudy) [inline tools/cpp/windows_cc_configure.bzl]: Why switching these two content? Now `%{msys_x64_mingw_content}` will actually be replaced with msys toolchain content.

Comment (rongjiecomputer) [reply tools/cpp/windows_cc_configure.bzl]: Those were the remnants of my previous idea. I forgot to change them back. Done.

**Comment by rongjiecomputer** [comment]
One instance of `toolchain_identifier: "msys_x64_mingw"` is renamed to `mingw_x64` since GCC/Mingw64 detection is no longer always tied to MSYS environment. I think the name change won't affect any existing user?

**Comment by meteorcloudy** [comment]
@rongjiecomputer Yes, the name changing should be fine.

**Comment by rongjiecomputer** [comment]
Please don't merge this PR. I found out that this will unintentionally break Unix platforms. I will update again if I have a better solution.

**Comment by meteorcloudy** [comment]
Hi @rongjiecomputer, we also realize `TERM` is a very common env that also exists on Linux. If you switch between different terminals under Linux, cc_configure will be rerun, that is not the ideal case.
Can you use `MSYSTEM` instead of `TERM` to tell if Bazel is running under MSYS?
The value is `MSYS` or `MINGW64` depending you are running under MSYS2 MSYS or MSYS2 MINGW64

File: tools/cpp/cc_configure.bzl [hunk 2]
```diff
@@ -79,6 +79,7 @@ cc_autoconf = repository_rule(
         "GCOV",
         "HOMEBREW_RUBY_PATH",
         "SYSTEMROOT",
+        "TERM", # Use to detect if Bazel is running in MSYS.
```
Comment (meteorcloudy) [inline tools/cpp/cc_configure.bzl]: Please considering use `MSYSTEM` instead of `TERM`

Comment (rongjiecomputer) [reply tools/cpp/cc_configure.bzl]: Thanks for the tips! TIL `msys2_shell.bat`, `mingw64_shell.bat` and `mingw32_shell.bat` are not entirely the same.

Comment (meteorcloudy) [reply tools/cpp/cc_configure.bzl]: Yes, IIRC they have different PATH value to make different gcc version available by default.

**Comment by rongjiecomputer** [comment]
Closing this as it diverges too far from the current shape of `tools/cpp/windows_cc_configure.bzl`. I will revive some of the idea of this PR when assisting #7320.

**Pull request closed without merging**